autoswitch_virtualenv.plugin.zsh

```diff
@@ -88,7 +88,7 @@ function _maybeworkon() {
     fi
 
     # Don't reactivate an already activated virtual environment
-    if [[ -z "$VIRTUAL_ENV" || "$venv_name" != "$(_get_venv_name $VIRTUAL_ENV $venv_type)" ]]; then
+    if [[ -z "$VIRTUAL_ENV" || "$venv_name" != "$(_get_venv_name $VIRTUAL_ENV $venv_type)" || "${+functions[deactivate]}" -eq 0 ]]; then
```

Comment:
could you add a code comment explaining this addition (something similar to your PR description would suffice)